src/engine/guidance/post_processing.cpp

```diff
@@ -1037,6 +1039,9 @@ std::vector<RouteStep> collapseTurns(std::vector<RouteStep> steps)
     // a series of turns is only possible to collapse if its only name changes and suppressed turns.
     const auto canCollapseAll = [&steps](std::size_t index, const std::size_t end_index) {
         BOOST_ASSERT(end_index <= steps.size());
+        if (!compatible(steps[index], steps[index + 1]))
+            return false;
+        ++index;
```

Comment:
Before we didn't check the very first `collapse` for compatibility.
What was tested before was for `a,b,c,d,e` you would check `b,c,d,e` since we are looping over all instructions unequal to `TurnType::Suppressed`.
Now we check `a,b` before doing the normal loop. This `+index` is compensated for by an additional `-1` for calling the function here.